src/vivarium_gates_iv_iron/components/observers.py

```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.configuration = builder.configuration.metrics.hemoglobin
+        self.start_time = get_time_stamp(builder.configuration.time.start)
+        self.hemoglobin = builder.value.get_value("hemoglobin.exposure")
+        self.step_size = builder.time.step_size()
+        self.age_bins = utilities.get_age_bins(builder)
+        self.exposure = Counter()
+
+        columns_required = ['alive', 'pregnancy_status', 'maternal_hemorrhage']
+        if self.configuration.by_age:
+            columns_required += ['age']
+        if self.configuration.by_sex:
+            columns_required += ['sex']
+        self.population_view = builder.population.get_view(columns_required)
+
+        builder.event.register_listener('collect_metrics', self.on_collect_metrics)
+        builder.value.register_value_modifier('metrics', self.metrics)
+
+    def on_collect_metrics(self, event: Event):
+        pop = self.population_view.get(event.index)
+        pop['hemoglobin'] = self.hemoglobin(event.index)
+        configuration = self.configuration.to_dict()
+        exposure_sum = {}
+
+        for p_state in models.PREGNANCY_MODEL_STATES:
+            for h_state in models.MATERNAL_HEMORRHAGE_STATES:
+                base_key = get_output_template(**configuration).substitute(measure=f'hemoglobin_exposure_sum_among_{p_state}_with_{h_state}',
+                                                                           year=event.time.year)
+                base_filter = QueryString(
+                    f'alive == "alive" and pregnancy_status == "{p_state}" and maternal_hemorrhage == "{h_state}"')
+                exposure_sum.update(get_group_counts(pop,
+                                                     base_filter, base_key,
+                                                     self.configuration,
+                                                     self.age_bins,
+                                                     aggregate=(lambda x: sum(x.hemoglobin))))
+
+        self.exposure.update(exposure_sum)
+
+    def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
+        metrics.update(self.exposure)
+        return metrics
+
+
+class AnemiaObserver:
+    configuration_defaults = {
+        'metrics': {
+            'anemia': {
+                'by_age': True,
+                'by_year': True,
+                'by_sex': False,
+            }
+        }
+    }
+
+    def __repr__(self):
+        return 'AnemiaObserver()'
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def sub_components(self) -> List:
+        return []
+
+    @property
+    def name(self):
+        return 'anemia_observer'
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
```

Comment:
You can use event.time.year, so you don't need a clock.

Reply:
Quick note, `event.time == self.clock() + event.step_size`, so if you remove `self.clock()`, you'd want to replace it with `event.time - event.step_size`. I"m not sure that makes sense upon reflection.

Reply:
I decided to keep the clock here. Thanks much for the other finds!